js/mailalerts.js

```diff
@@ -33,20 +33,39 @@ function  addNotification() {
     success: function (resp) {
       resp = JSON.parse(resp);
 
-      $('div.js-mailalert > span').html('<article class="alert alert-info" role="alert" data-alert="success">'+resp.message+'</article>').show();
+      const alertProperties = {
+        class: resp.error ? 'danger' : 'success',
+        data: resp.error ? 'error' : 'success'
+      }
+
+      $('div.js-mailalert > span').html('<article class="mt-1 alert alert-' + alertProperties.class + '" role="alert" data-alert="' + alertProperties.data + '">'+resp.message+'</article>').show();
       if (!resp.error) {
         $('div.js-mailalert > button').hide();
         $('div.js-mailalert > input[type=email]').hide();
-        $('div.js-mailalert > #gdpr_consent').hide();
+        $('div.js-mailalert .gdpr_consent_wrapper').hide();
       }
     }
   });
   return false;
 }
 
-$('document').ready(function()
-{
-  $('.js-remove-email-alert').click(function()
+$(document).on('ready', function() {
+  const mailAlertWrapper = $('.js-mailalert');
+  const mailAlertSubmitButton = mailAlertWrapper.find('button');
+
+  if (mailAlertWrapper.find('#gdpr_consent').length) {
```

Comment:
```suggestion
  if ($find('#gdpr_consent').length) {
```
It's an id, ids are unique so directly fetch it :)

Reply:
@PierreRambaud I don't belive third-party modules, this is why it's more precise :P

Reply:
We shouldn't think about third-party modules, they have to use the correct HTML structure. We shouldn't do something illogic

Reply:
@PierreRambaud I understand but we're talking about PrestaShop module here, and entire PR is because of lack of the compatibility between those two.

Reply:
I know, but the more we add bad practices usage, the fewer modules developers will take time to do it.